### PR TITLE
Community of origin report

### PR DESCRIPTION
**Pull request by eanders**
## Description

Provides the community of origin report.

## Type of change
- [ ] Bug fix
- [x] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

**Comment by eanders** [comment]
Example PDF
[Community of Origin 2023-11-03 17 14 49.pdf](https://github.com/greenriver/hmis-warehouse/files/13254748/Community.of.Origin.2023-11-03.17.14.49.pdf)

File: app/models/grda_warehouse/shape/shared_behaviors.rb [hunk 1]
```diff
@@ -26,9 +26,10 @@ def geo_json_properties
         }.merge(additional_geo_json_properties)
       end
 
-      def name
-        'unknown'
-      end
+      # why?
+      # def name
+      #   'unknown'
+      # end
```
Comment (awisema) [inline app/models/grda_warehouse/shape/shared_behaviors.rb]: Remove, or explain?

File: drivers/boston_reports/app/models/boston_reports/community_of_origin.rb [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+      GrdaWarehouse::Shape.geo_collection_hash(GrdaWarehouse::Shape::ZipCode.where(zcta5ce10: zips))
+    end
+
+    def zip_code_colors
+      [
+        { color: '#BF216B', range: [0.02] },
+        { color: '#F22797', range: [0.02, 0.05] },
+        { color: '#F2BC1B', range: [0.05, 0.1] },
+        { color: '#F26A1B', range: [0.1, 0.15] },
+        { color: '#F5380E', range: [0.15] },
```
Comment (awisema) [inline drivers/boston_reports/app/models/boston_reports/community_of_origin.rb:223]: Maybe use Ruby ranges here? I was confused by `range: [0.02]` and maybe `(.. 0.02)` would be clearer?

File: drivers/boston_reports/app/models/boston_reports/community_of_origin.rb [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+      ]
+    end
+
+    def detail_headers
```
Comment (awisema) [inline drivers/boston_reports/app/models/boston_reports/community_of_origin.rb:228]: Maybe remove?

**Review by awisema — APPROVED** [review]
Looks good! A couple of minor readability tweaks.

File: drivers/boston_reports/app/models/boston_reports/community_of_origin.rb (hunk 2 above)
Comment (eanders) [reply drivers/boston_reports/app/models/boston_reports/community_of_origin.rb:223]: Thanks, leaving, but added note, this is used by JavaScript, converting to ranges would probably break that and require a JS re-write.

File: drivers/boston_reports/app/models/boston_reports/community_of_origin.rb (hunk 3 above)
Comment (eanders) [reply drivers/boston_reports/app/models/boston_reports/community_of_origin.rb:228]: Thanks, noted that we're leaving that temporarily, I think there's a non-zero chance we'll end up building a detail page, this felt like a useful pattern.

**Comment by ttoomey** [comment]
Hi @eanders - just fyi looks like this didn't get squashed before merging into release-92

**Comment by eanders** [comment]
Yep, I noticed that as well, but didn't think it worth trying to cleanup.  Let me know if you think we should.

**Pull request merged**